src/__tests__/JWT-test.ts

```diff
@@ -297,14 +299,56 @@ describe('verifyJWT()', () => {
     })
   })
 
-  it('accepts a valid nbf', () => {
-    return createJWT({ nbf: NOW + NBF_SKEW }, { issuer: did, signer }).then(
-      jwt =>
-        verifyJWT(jwt).then(
-          ({ payload }) => expect(payload).toMatchSnapshot(),
-          error => expect(error).toBeNull()
-        )
-    )
+  describe('validFrom timestamp', () => {
+    it('passes when nbf is in the past', async () => {
+      const jwt = await createJWT(
+        { nbf: PAST },
+        { issuer: did, signer }
+      )
+      expect(verifyJWT(jwt)).resolves.not.toThrow()
+    })
+    it('passes when nbf is in the past and iat is in the future', async () => {
+      const jwt = await createJWT(
+        { nbf: PAST, iat: FUTURE },
+        { issuer: did, signer }
+      )
+      expect(verifyJWT(jwt)).resolves.not.toThrow()
+    })
+    it('fails when nbf is in the future', async () => {
+      const jwt = await createJWT(
+        { nbf: FUTURE },
+        { issuer: did, signer }
+      )
+      expect(verifyJWT(jwt)).rejects.toThrow()
+    })
+    it('fails when nbf is in the future and iat is in the past', async () => {
+      const jwt = await createJWT(
+        { nbf: FUTURE, iat: PAST },
+        { issuer: did, signer }
+      )
+      expect(verifyJWT(jwt)).rejects.toThrow()
+    })
+    it('passes when nbf is missing and iat is in the past', async () => {
+      const jwt = await createJWT(
+        { iat: PAST },
+        { issuer: did, signer }
+      )
+      expect(verifyJWT(jwt)).resolves.not.toThrow()
+    })
+    it('fails when nbf is missing and iat is in the future', async () => {
+      const jwt = await createJWT(
+        { iat: FUTURE },
+        { issuer: did, signer }
+      )
+      expect(verifyJWT(jwt)).rejects.toThrow()
+    })
+    it('passes when nbf and iat are both missing', async () => {
+      const jwt = await createJWT(
+        {},
```

Comment:
The jwt created here still has an `iat` field added by default.
We should probably use `"iat" : undefined` in the payload to force it to be removed, or even better, use hard-coded JWTs for these tests.

Reply:
fixed